streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -303,18 +303,32 @@ abstract class ZSink[-R, +E, -I, +L, +Z] private (
     } yield push)
 
   /**
-   * Returns the sink that executes this one and times its execution.
+   * Returns a sink that summarizes an effect by computing some value before and
+   * after execution, and then combining the values to produce a summary,
+   * together with the result of the sink.
    */
-  final def timed: ZSink[R with Clock, E, I, L, (Z, Duration)] =
+  final def summarized[R1 <: R, B, C](summary: ZIO[R1, Nothing, B])(f: (B, B) => C): ZSink[R1, E, I, L, (Z, C)] =
```

Comment:
@adamgfraser  I have to fix the error channel to `Nothing` in `summary: ZIO[R1, Nothing, B]`. It's different from that in `ZIO#summarized`, which uses `E1 >: E`. In `ZSink`, the error channel is a more complex data structure (something like `(Either[E, (Z, B, B)], Chunk[L])`). If `E1` is used, the super type ends up being `Any`.

It'd be nice to still use `E1` (instead of `Nothing`) but somehow express the type relationship between `E1` and `(Either[E, (Z, B, B)], Chunk[L])`. Do you have any suggestion?

Reply:
Yes this should definitely support an error type. The error type of the workflow would need to be mapped to a `Left` and an empty `Chunk` for this to unify.

However, as this operator is already implemented in `series/2.x` I wonder if it just makes sense to wait for that.

Reply:
Unfortunately, will need it in 1.x :)

I'll try to fix the error type.

Reply:
Been trying to fix the error type but so far haven't much luck to get `(Either[E1, (Z, C)], Chunk[L])` to work unfortunately.